server.js

```diff
@@ -1,34 +1,67 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
-
-// Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
-
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
-
-// Start the server
+import express from "express";
+import cors from "cors";
+import goldenGlobesData from "./data/golden-globes.json";
+
+const port = process.env.PORT || 8080;
+const app = express();
+app.use(cors());
+
+app.get("/", (request, response) => {
+  const dataLength = String(goldenGlobesData.length);
+  response.send(
+    `You can find ${dataLength} movies from Golden Globes ceremony here!`
+  );
+});
+
+app.get("/golden-globes/ceremonies", (request, response) => {
+  const ceremoniesDuplicated = goldenGlobesData.map((movie) => movie.ceremony);
+
+  const ceremoniesUnique = [];
+  ceremoniesDuplicated.forEach((item) => {
+    if (!ceremoniesUnique.includes(item)) {
+      ceremoniesUnique.push(item);
+    }
+  });
+
+  res.json({ data: ceremoniesUnique });
+});
+
+app.get("/golden-globes", (request, response) => {
+  const { ceremony, win, category } = request.query;
+  let result = goldenGlobesData;
+
+  if (ceremony) {
+    result = goldenGlobesData.filter(
+      (movie) => movie.ceremony === Number(ceremony)
+    );
+  }
+
+  if (win) {
+    result = result.filter((movie) => movie.win);
+  }
+
+  if (category) {
+    result = result.filter((movie) => movie.category.includes(category));
+  }
+  response.json(result);
+});
+
+app.get("/golden-globes/:film", (request, response) => {
+  const { film } = request.params;
+  const filteredFilm = goldenGlobesData.filter((movie) => {
+    const currentFilm = movie.film.toString().toLowerCase();
+    return currentFilm.includes(film.toLowerCase());
+  });
+
+  if (filteredFilm.length > 0) {
+    response.json(filteredFilm);
+  } else {
+    response
+      .status(404)
+      .json({ message: `There is not information about ${film}` });
+  }
+});
+
```

Comment:
To meet blue requirments you need to provide:
- A minimum of one endpoint to return a **single** result (single element).

Reply:
This will be returned one film. Actually, the golden globe data doesn't have any other unique value.

Reply:
It is not a single result (you have 3 records here) You need to provide an endpoint which returns always 1 record. ![image](https://user-images.githubusercontent.com/50926159/124393082-098af080-dcf9-11eb-9638-60f547727b34.png)

Reply:
Thanks for putting time into this :)
Let me explain it a bit more about why my API cannot return unique result. So as you can see the data structure provided by Technigo:
```
{
    "year_film": 2009,
    "year_award": 2010,
    "ceremony": 67,
    "category": "Best Performance by an Actress in a Motion Picture - Musical or Comedy",
    "nominee": "Meryl Streep",
    "film": "Julie & Julia",
    "win": true
}
```
There is no field that can be unique because there is no such a `filmId` here.
`year_film` => Many films can be produced in a year.
`year_award` => Many films can be awarded in a year.
`ceremony` => It's very clear that it cannot be a unique field.
`category` => Many films can be under a category. We have many comedy films.
`nominee` => I think your image can explain that it cannot be unique.
`film` => Is the film name and based on the image you sent it cannot be unique.
`win` => It's clear that it cannot be unique.
Based on what I explained. I thought that the most probable unique field can be `film`. But I can see that is not.
I believe the data structure is not proper enough to have a single endpoint although you have any suggestions about the unique field I would be really happy to implement that :)
Btw if there was an `id` property the code would be like this:
```js
app.get("/golden-globes/:id", (request, response) => {
  const { id } = request.params;
  const foundFilm = goldenGlobesData.find((movie) => {
    return movie.id === id;
  });

  if (foundFilm) {
    response.json(foundFilm);
  } else {
    response
      .status(404)
      .json({ message: `No film has found with this id: ${id}` });
  }
});
```